crates/matrix-sdk-crypto/src/identities/manager.rs

```diff
@@ -1578,6 +1583,43 @@ pub(crate) mod tests {
         );
     }
 
+    #[async_test]
+    async fn test_is_user_tracked_and_up_to_date() {
+        let manager = manager_test_helper(user_id(), device_id()).await;
+        let bob = user_id!("@bob:example.org");
+
+        let cache = manager.store.cache().await.unwrap();
+        let key_query_manager = manager.key_query_manager.synced(&cache).await.unwrap();
+
+        let users = std::iter::once(bob);
+        manager.update_tracked_users(users.clone()).await.unwrap();
+
+        assert!(key_query_manager.tracked_users().contains(bob), "Bob should now be tracked");
+        assert!(
+            key_query_manager.users_for_key_query().await.0.contains(bob),
+            "Bob should now be queued up for a /keys/query"
+        );
+        assert!(
+            !key_query_manager.is_user_tracked_and_up_to_date(bob).await,
+            "Bob should now be tracked but also dirty"
+        );
+
+        key_query_manager
+            .mark_tracked_users_as_up_to_date(users, SequenceNumber::default())
+            .await
+            .unwrap();
+
+        assert!(key_query_manager.tracked_users().contains(bob), "Bob should still be tracked");
+        assert!(
+            !key_query_manager.users_for_key_query().await.0.contains(bob),
+            "Bob should'nt be queued up for a /keys/query anymore"
```

Comment:
```suggestion
            "Bob shouldn't be queued up for a /keys/query any more"
```

(though apparently the anymore/any more thing might be a British vs American English thing. Whatever. "should'nt" is definitely wrong though :) )

Reply:
Reverted.